### PR TITLE
msg: addr -> addrvec (part 1)

### PR DESCRIPTION
**Pull request by liewegas**
- [x] Initial stubs for entity_addrvec_t instead of entity_addr_t in teh Messenger interface
- [x] MgrMap conversion
- [x] MDSMap conversion
- [x] OSDMap conversion

The final monmap change will come after the messenger(s) actually support multi-bind etc.

### CODE REVIEW

File: src/mon/MgrMap.h [hunk 1]
```diff
@@ -132,7 +132,7 @@ class MgrMap
   /// global_id of the ceph-mgr instance selected as a leader
   uint64_t active_gid = 0;
   /// server address reported by the leader once it is active
```
Comment (rjfd) [inline src/mon/MgrMap.h]: nit: should we also update the comment string accordingly?

**Comment by rjfd** [comment]
@liewegas could you briefly explain why we will need to support a vector of addresses instead of a single address?

**Comment by liewegas** [comment]
@rjfd Anywhere we have an entity_addr_t today (well, most places... monmap, osdmap, mgrmap, etc.) we need to store both a legacy msgr1 address and a new msgr2 address.  It will also enable ipv4 and ipv6 dual-hosting.  The messengers will bind multiple times and answer to multiple addresses.

entity_addrvec_t and entity_addr_t already encode such that they are interchangeable (e.g., old code will decode an entity_addrvec_t as the first legacy addr in the list).

**Comment by rjfd** [comment]
@liewegas thanks!

**Comment by tchaikov** [comment]
@liewegas needs rebase.

File: src/messages/MOSDMarkMeDown.h [hunk 2]
```diff
@@ -55,8 +56,22 @@ class MOSDMarkMeDown : public PaxosServiceMessage {
...
+	     payload, features);
+      encode(epoch, payload);
+      encode(request_ack, payload);
+      return;
+    }
+    header.version = HEAD_VERSION;
+    header.compat_version = COMPAT_VERSION;
     encode(fsid, payload);
     encode(target_osd, payload, features);
+    encode(target_addrs, payload, features);
```
Comment (tchaikov) [inline src/messages/MOSDMarkMeDown.h]: should update `decode_payload()` accordingly.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-06-02_12:29:35-rados-wip-kefu-testing-2018-06-02-1754-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2018-06-02_16:38:39-rados-wip-kefu-testing-2018-06-02-1754-distro-basic-smithi/#

**Pull request merged**